src/fontcache/freetypefontcache.cpp

```diff
@@ -0,0 +1,334 @@
+	uint width  = std::max(1U, (uint)slot->bitmap.width + (this->fs == FS_NORMAL));
+	uint height = std::max(1U, (uint)slot->bitmap.rows  + (this->fs == FS_NORMAL));
+
+	/* Limit glyph size to prevent overflows later on. */
+	if (width > MAX_GLYPH_DIM || height > MAX_GLYPH_DIM) usererror("Font glyph is too large");
+
+	/* FreeType has rendered the glyph, now we allocate a sprite and copy the image into it */
+	SpriteLoader::Sprite sprite;
+	sprite.AllocateData(ZOOM_LVL_NORMAL, width * height);
+	sprite.type = ST_FONT;
+	sprite.colours = (aa ? SCC_PAL | SCC_ALPHA : SCC_PAL);
+	sprite.width = width;
+	sprite.height = height;
+	sprite.x_offs = slot->bitmap_left;
+	sprite.y_offs = this->ascender - slot->bitmap_top;
+
+	/* Draw shadow for medium size */
+	if (this->fs == FS_NORMAL && !aa) {
+		for (uint y = 0; y < (uint)slot->bitmap.rows; y++) {
+			for (uint x = 0; x < (uint)slot->bitmap.width; x++) {
+				if (HasBit(slot->bitmap.buffer[(x / 8) + y * slot->bitmap.pitch], 7 - (x % 8))) {
+					sprite.data[1 + x + (1 + y) * sprite.width].m = SHADOW_COLOUR;
+					sprite.data[1 + x + (1 + y) * sprite.width].a = 0xFF;
+				}
+			}
+		}
+	}
+
+	for (uint y = 0; y < (uint)slot->bitmap.rows; y++) {
+		for (uint x = 0; x < (uint)slot->bitmap.width; x++) {
+			if (aa ? (slot->bitmap.buffer[x + y * slot->bitmap.pitch] > 0) : HasBit(slot->bitmap.buffer[(x / 8) + y * slot->bitmap.pitch], 7 - (x % 8))) {
+				sprite.data[x + y * sprite.width].m = FACE_COLOUR;
+				sprite.data[x + y * sprite.width].a = aa ? slot->bitmap.buffer[x + y * slot->bitmap.pitch] : 0xFF;
+			}
+		}
+	}
+
+	GlyphEntry new_glyph;
+	new_glyph.sprite = BlitterFactory::GetCurrentBlitter()->Encode(&sprite, SimpleSpriteAlloc);
+	new_glyph.width  = slot->advance.x >> 6;
+
+	this->SetGlyphPtr(key, &new_glyph);
+
+	return new_glyph.sprite;
+}
+
+
+GlyphID FreeTypeFontCache::MapCharToGlyph(WChar key)
+{
+	assert(IsPrintable(key));
+
+	if (key >= SCC_SPRITE_START && key <= SCC_SPRITE_END) {
+		return this->parent->MapCharToGlyph(key);
+	}
+
+	return FT_Get_Char_Index(this->face, key);
+}
+
+const void *FreeTypeFontCache::InternalGetFontTable(uint32 tag, size_t &length)
+{
+	FT_ULong len = 0;
+	FT_Byte *result = nullptr;
+
+	FT_Load_Sfnt_Table(this->face, tag, 0, nullptr, &len);
+
+	if (len > 0) {
+		result = MallocT<FT_Byte>(len);
+		FT_Load_Sfnt_Table(this->face, tag, 0, result, &len);
+	}
+
+	length = len;
+	return result;
+}
+
+/**
+ * Free everything allocated w.r.t. freetype.
+ */
+void UninitFreeType()
+{
+	FT_Done_FreeType(_library);
+	_library = nullptr;
+}
+
+#endif /* WITH_FREETYPE */
+
+
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(WITH_FONTCONFIG) && !defined(WITH_COCOA)
+
+#ifdef WITH_FREETYPE
+FT_Error GetFontByFaceName(const char *font_name, FT_Face *face) { return FT_Err_Cannot_Open_Resource; }
+#endif /* WITH_FREETYPE */
+
+#endif /* !defined(_WIN32) && !defined(__APPLE__) && !defined(WITH_FONTCONFIG) && !defined(WITH_COCOA) */
```

Comment:
Why separate this from the main WITH_FREETYPE block?

Reply:
I remembered cleaning this bit up, but that must have been in a slightly different iteration!